src/main/java/com/exadel/sober/services/UserService.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.sober.services;
+
+import com.exadel.sober.models.User;
+import com.exadel.sober.repositories.PromiseRepository;
+import com.exadel.sober.repositories.UserRepository;
+import com.exadel.sober.repositoriesImplemetation.AddictionRepositoryImpl;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+
+@Service
+public class UserService {
+
+    private UserRepository userRepository;
+
+    private PromiseService promiseService;
+
+    public UserService(UserRepository userRepository,
+                       PromiseService promiseService) {
+        this.userRepository = userRepository;
+        this.promiseService = promiseService;
+    }
+
+    public User addNewUser(User newUser) {
```

Comment:
don't forget about validation because you even don't know if User has email and password of if they are valid